src/main/kotlin/org/rust/ide/intentions/RemoveDbgIntention.kt

```diff
@@ -0,0 +1,50 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.intentions
+
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiDocumentManager
+import com.intellij.psi.PsiElement
+import org.rust.lang.core.psi.RsMacroExpr
+import org.rust.lang.core.psi.ext.ancestorStrict
+import org.rust.lang.core.psi.ext.macroName
+import org.rust.lang.core.psi.ext.startOffset
+
+/**
+ * Removes the `dbg!` macro
+ *
+ * ```
+ * let a = dbg!(variable);
+ * ```
+ *
+ * to this:
+ *
+ * ```
+ * let a = variable;
+ * ```
+ */
+class RemoveDbgIntention : RsElementBaseIntentionAction<RsMacroExpr>() {
+
+    override fun getText() = "Remove dbg!"
+    override fun getFamilyName() = text
+
+    override fun findApplicableContext(project: Project, editor: Editor, element: PsiElement): RsMacroExpr? {
+        val macroExpr = element.ancestorStrict<RsMacroExpr>() ?: return null
+        if (macroExpr.macroCall.macroName != "dbg") {
+            return null
+        }
+        return macroExpr
+    }
+
+    override fun invoke(project: Project, editor: Editor, ctx: RsMacroExpr) {
+        val expr = ctx.macroCall.exprMacroArgument?.expr ?: return
+        val cursorOffsetToExpr = editor.caretModel.offset - expr.startOffset
```

Comment:
Looks like it doesn't work in such cases:
`foobar(d/*caret*/bg!(123));`
`foobar(dbg!(123   <long whitespace>   /*caret*/));`

Try use `kotlin.math.min/max` like this:
```
val cursorOffsetToExpr = max(0, editor.caretModel.offset - expr.startOffset)
val resultOffset = min(newExpr.startOffset + cursorOffsetToExpr, newExpr.endOffset)
```
and add such tests